src/compiler/types.ts

```diff
@@ -1099,6 +1118,13 @@ namespace ts {
         _statementBrand: any;
     }
 
+    // Represents a statement that is elided as part of a transformation to emit comments on a
+    // not-emitted node.
+    // @internal
+    // @kind(SyntaxKind.NotEmittedStatement)
+    export interface NotEmittedStatement extends Statement {

```

Comment:
That would be like an `EmptyStatement` and would add an unwanted extra semicolon. Using a `NotEmittedStatement` also reduces the number of unnecessary allocations.